spring-cloud-consul-discovery/src/main/java/org/springframework/cloud/consul/discovery/ConsulDiscoveryProperties.java

```diff
@@ -86,7 +86,12 @@
 	 * Use ip address rather than hostname during registration
 	 */
 	private boolean preferIpAddress = false;
-
+	
+	/**
+	 * Source of how we will determine the address to use
+	 */
+	private AddressSource addressSource = AddressSource.AGENT;
```

Comment:
I was thinking enum also because we could basically have the values like AGENT, IP, HOSTNAME at that point and possibly simplify the preferIpAddress also.  Would you rather me roll all that support up as well into the enum so the value is more apparent?